x-pack/plugins/rollup/public/crud_app/sections/job_list/job_table/job_table.test.js

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Pager } from '@elastic/eui';
+
+import { registerTestBed } from '../../../../../__jest__/utils';
+import { getJobs } from '../../../../../fixtures';
+import { rollupJobsStore } from '../../../store';
+import { JobTable } from './job_table';
+
+const defaultProps = {
+  jobs: [],
+  pager: new Pager(20, 10, 1),
+  filter: '',
+  sortField: '',
+  isSortAscending: false,
+  openDetailPanel: () => {},
+  closeDetailPanel: () => {},
+  filterChanged: () => {},
+  pageChanged: () => {},
+  pageSizeChanged: () => {},
+  sortChanged: () => {},
+};
+
+const initTestBed = registerTestBed(JobTable, defaultProps, rollupJobsStore);
+
+describe('<JobTable />', () => {
+  describe('table rows', () => {
+    const totalJobs = 5;
+    const jobs = getJobs(totalJobs);
+    const openDetailPanel = jest.fn();
+    const { findTestSubject } = initTestBed({ jobs, openDetailPanel });
+    const tableRows = findTestSubject('jobTableRow');
+
+    it('should create 1 table row per job', () => {
+      expect(tableRows.length).toEqual(totalJobs);
+    });
+
+    it('should create the expected 8 columns for each row', () => {
+      const expectedColumns = [
+        'id',
+        'status',
+        'indexPattern',
+        'rollupIndex',
+        'rollupDelay',
+        'dateHistogramInterval',
+        'groups',
+        'metrics'
+      ];
+
+      const tableColumns = expectedColumns
```

Comment:
I think it might be a little easier to follow this if we use `reduce`. What do you think of this?

```js
const tableColumns = expectedColumns.reduce((foundColumns, columnId) => {
  const column = findTestSubject(`jobTableHeaderCell-${columnId}`);
  if (column.length > 0) {
    return foundColumns.concat(columnId);
  }
  return foundColumns;
}, []);
```

Reply:
Exactly, thanks! 😊 That's one of those moments when your brain slows down and you just do things more complex than necessary. But I know that I have you to double check that!